crates/starlight/src/vm/value.rs

```diff
@@ -1384,13 +1386,14 @@ pub mod new_value {
 
         #[inline]
         pub fn get_object(self) -> GcPointer<dyn GcCell> {
-            assert!(self.is_pointer());
+            assert!(self.is_object());
+            
             unsafe { std::mem::transmute(self.0.ptr) }
         }
 
         #[inline]
         pub fn is_object(self) -> bool {
-            self.is_pointer()
+            self.is_pointer() && !self.is_bool() && !self.is_undefined() && !self.is_null() && !self.is_empty()
```

Comment:
Just !self.is_empty is enough

Reply:
`is_bool` and `is_undefined` and `is_null` should not considered to `object` ? if then, when serializing or deserializing, and other occasions, we will consider it a legal GcPointer?

Reply:
is_object is just alias to is_pointer plus a check that it is not a null pointer.

Reply:
Just leave is_pointer() and !is_empty() checks, other than that I do not see any mistakes in the code

Reply:
but `undefined`, `bool`, `null` etc should differ from real pointer?
For example, here https://github.com/Starlight-JS/starlight/blob/154834e945b6e24d828133af2eed3c6f35c5b9d4/crates/starlight/src/gc/snapshot/serializer.rs#L386-L397, when it's real pointer, we should write GcPointer ?

Reply:
there's no pointer out there. is_pointer exists only for backwards compatibility with `old_value` module which has separate tags for string and object although these tags are not used.

Reply:
but then , undefined,null will meet is_object and write GcPointer, which should be write value directly?

Reply:
they will not meet it. You shall read doc for JsValue more precisely:
```
* The tag 0x0000 denotes a pointer, or another form of tagged immediate. Boolean,
         * null and undefined values are represented by specific, invalid pointer values:
```
https://github.com/Starlight-JS/starlight/blob/dev/crates/starlight/src/vm/value.rs#L1219

Reply:
Oh I see. I will change it.